contrib/pyln-client/tests/test_plugin.py

```diff
@@ -362,3 +362,25 @@ def test1(msat: Millisatoshi):
 
     ba = p._bind_pos(test1, ["100msat"], None)
     test1(*ba.args, **ba.kwargs)
+
+
+def test_duplicate_result():
+    p = Plugin(autopatch=False)
+
+    def test1(request):
+        request.set_result(1)
+        request.set_result(1)
+
+    req = Request(p, req_id=1, method="test1", params=[])
+    ba = p._bind_kwargs(test1, {}, req)
+    with pytest.raises(ValueError, match=r'current state is RequestState.FINISHED'):
+        test1(*ba.args)
+
+    def test2(request):
+        request.set_exception(1)
+        request.set_exception(1)
+
+    req = Request(p, req_id=1, method="test2", params=[])
+    ba = p._bind_kwargs(test2, {}, req)
+    with pytest.raises(ValueError, match=r'current state is RequestState.FAILED'):
+        test2(*ba.args)
```

Comment:
I would improve on the testcase like in db7f6cdf9  
 - more cases: set_exception before set_result and other way around
 - checks for the occurrence of the stack with a MARKER comment